GH Actions: Run validatePullRequest and MiMa

marcospereira opened the pull request:
## Changes

- GH Actions: run validatePullRequest instead of just compiling the tests
- Run MiMa checks on GH Actions instead of Travis CI

## Caveat

No tests will be run in this PR since `validatePullRequest` has no changes to detect in the actual Scala/Java code. We will see the results in the upcoming PRs after this got merged.

## After merging changes

- Disable akka/akka builds on Travis CI.

marcospereira committed:
GH Actions: run validatePullRequest instead of just compiling the tests

marcospereira committed:
Run MiMa checks on GH Actions instead of Travis CI

marcospereira committed:
Some clear names for workflow and steps

marcospereira commented on .github/workflows/build-test-prValidation.yml:
```diff
@@ -43,10 +41,10 @@ jobs:
           -Dsbt.log.noformat=false \
           -Dakka.log.timestamps=true \
-          scalafmtCheckAll scalafmtSbtCheck
+          scalafmtCheckAll scalafmtSbtCheck headerCheckAll
```
This was running on Travis CI before, together with MiMa reports. I think it is better to have it as part of the code style checks, though.

marcospereira commented on .github/workflows/build-test-prValidation.yml:
```diff
@@ -111,7 +108,32 @@ jobs:
+          java-version: adopt@1.8.0
+
+      - name: Report MiMa Binary Issues
+        run: sbt -jvm-opts .jvmopts-ci -Dakka.build.scalaVersion=${{ matrix.scalaVersion }} ";update ;mimaReportBinaryIssues ;Test/compile ;validateCompile"
```
I'm not sure we need `Test/compile` here, but it was part of Travis CI. Also, maybe move `validateCompile` to the job running `validatePullRequest`?

patriknw replied:
I agree, compiling everything should be covered by other steps

also, that `update` shouldn't be needed?

marcospereira committed:
Add cache to bin compat step

patriknw commented on .github/workflows/build-test-prValidation.yml:
```diff
@@ -89,16 +87,15 @@ jobs:
         uses: olafurpg/setup-scala@v10
         with:
-          java-version: adopt@1.8.0
+          java-version: adopt@1.11
```
didn't we use 8 for pr validation in Jenkins? it can be pretty good to catch such mistakes early in validation, since we support 8 but all of us use 11 for development

marcospereira committed:
Also run MiMa for master branch and tags

patriknw reviewed (commented):
good, just some small comments

marcospereira committed:
Use JDK 8 to validate pull request

marcospereira committed:
Move validateCompile to pull request validation

Also remove `update` and `Test/compile` from binary compatibility checks:

- `validateCompile` runs for all configurations, so it overrides `Test/compile`
- `update` is unnecessary for mima checks

marcospereira commented:
@patriknw some additional changes:

- Add a separated MiMa workflow that now also runs for `master` and 2.6 tags
- Using JDK 8 to run on pull request validation job
- Remove some unnecessary tasks (`Test/compile`, `update`) and moved `validateCompile` to pull request validate job

This should be good to go now. :-)

patriknw commented:
@marcospereira Do we need to remove Travis from the branch settings now, or is it just slow? waiting for status.

patriknw reviewed (approved):
LGTM

johanandren reviewed (approved):
LGTM
